x/ccv/provider/keeper/grpc_query.go

```diff
@@ -201,7 +201,7 @@ func (k Keeper) QueryValidatorProviderAddr(goCtx context.Context, req *types.Que
 	}
 	consumerAddr := types.NewConsumerConsAddress(consumerAddrTmp)
 
-	providerAddr, found := k.GetValidatorByConsumerAddr(ctx, req.ChainId, consumerAddr)
+	providerAddr, found := k.GetValidatorByConsumerAddr(ctx, req.ConsumerId, consumerAddr)
```

Comment:
Nice catch. @insumity why do we have deprecated fields in query requests. Given that both chanId and consumerId have the same type, it's easy to have bugs where we call functions with a chainId.

Reply:
For backward compatibility. We try to prevent `chainIds` from being used by doing:
```
} else if req.ChainId != "" {
    return nil, status.Errorf(codes.InvalidArgument, "ChainId has been deprecated. Use ConsumerId instead.")
}
```
But in this case we unfortunately missed it.

Reply:
> For backward compatibility.

Can you please give more details about this. What's the use case that we want to keep compatibility with?

Reply:
We could imagine that Hermes is using this query (`QueryValidatorProviderAddr`) so they use our protos to build Hermes. If we were to simply go and remove `chain_id`, we would break their builds. It seems, in those cases, the right approach is to just mark `chain_id` as deprecated. It is true that the actual query is still broken from their side but this seems less severe.
We could remove all those `chain_id`s if we were to bump the proto version to `v2` but this seems like it would be more time consuming at this stage.

Reply:
Bumping the proto version is something we should talk to @MSalopek. 

Regarding breaking the queries for Hermes for example, as you stated it will be broken anyway. All the users of these queries will need to update their code. Leaving chainId there seems even more dangerous to me as there is a chance that we forget to return an error and the query goes through and returns weird stuff. So my recommendation would be to remove deprecated fields from the queries.

Reply:
I would avoid `v2` and inform hermes that there will be a compatibility break and that they should update their code.